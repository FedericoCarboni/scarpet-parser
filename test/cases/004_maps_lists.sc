// Maps and lists
{
    'key1' -> 'value1',
    'key2' -> 'value2',
};
{);
[
    'value1', 'value2',
];
[);
