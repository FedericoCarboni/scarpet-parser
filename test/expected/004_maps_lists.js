export const diagnostics = [
    {
        code: 10,
        range: {
            start: {
                offset: 72,
                line: 5,
                character: 1,
            },
            end: {
                offset: 73,
                line: 5,
                character: 2,
            },
        },
    },
    {
        code: 10,
        range: {
            start: {
                offset: 105,
                line: 9,
                character: 1,
            },
            end: {
                offset: 106,
                line: 9,
                character: 2,
            },
        },
    },
];

/** @type {import('../../lib/Parser.js').Node} */
export const root = {
    kind: 'BinaryExpression',
    operator: ';',
    lvalue: {
        kind: 'BinaryExpression',
        operator: ';',
        lvalue: {
            kind: 'BinaryExpression',
            operator: ';',
            lvalue: {
                kind: 'BinaryExpression',
                operator: ';',
                lvalue: {
                    kind: 'MapLiteral',
                    params: [
                        {
                            kind: 'BinaryExpression',
                            operator: '->',
                            lvalue: {
                                kind: 'StringLiteral',
                                literal: "'key1'",
                                value: 'key1',
                                start: {
                                    offset: 24,
                                    line: 2,
                                    character: 4,
                                },
                                end: {
                                    offset: 30,
                                    line: 2,
                                    character: 10,
                                },
                            },
                            rvalue: {
                                kind: 'StringLiteral',
                                literal: "'value1'",
                                value: 'value1',
                                start: {
                                    offset: 34,
                                    line: 2,
                                    character: 14,
                                },
                                end: {
                                    offset: 42,
                                    line: 2,
                                    character: 22,
                                },
                            },
                            comment: null,
                            start: {
                                offset: 24,
                                line: 2,
                                character: 4,
                            },
                            end: {
                                offset: 42,
                                line: 2,
                                character: 22,
                            },
                        },
                        {
                            kind: 'BinaryExpression',
                            operator: '->',
                            lvalue: {
                                kind: 'StringLiteral',
                                literal: "'key2'",
                                value: 'key2',
                                start: {
                                    offset: 48,
                                    line: 3,
                                    character: 4,
                                },
                                end: {
                                    offset: 54,
                                    line: 3,
                                    character: 10,
                                },
                            },
                            rvalue: {
                                kind: 'StringLiteral',
                                literal: "'value2'",
                                value: 'value2',
                                start: {
                                    offset: 58,
                                    line: 3,
                                    character: 14,
                                },
                                end: {
                                    offset: 66,
                                    line: 3,
                                    character: 22,
                                },
                            },
                            comment: null,
                            start: {
                                offset: 48,
                                line: 3,
                                character: 4,
                            },
                            end: {
                                offset: 66,
                                line: 3,
                                character: 22,
                            },
                        },
                        null,
                    ],
                    start: {
                        offset: 18,
                        line: 1,
                        character: 0,
                    },
                    end: {
                        offset: 69,
                        line: 4,
                        character: 1,
                    },
                },
                rvalue: {
                    kind: 'MapLiteral',
                    params: [null],
                    start: {
                        offset: 71,
                        line: 5,
                        character: 0,
                    },
                    end: {
                        offset: 73,
                        line: 5,
                        character: 2,
                    },
                },
                comment: null,
                start: {
                    offset: 18,
                    line: 1,
                    character: 0,
                },
                end: {
                    offset: 73,
                    line: 5,
                    character: 2,
                },
            },
            rvalue: {
                kind: 'ListLiteral',
                params: [
                    {
                        kind: 'StringLiteral',
                        literal: "'value1'",
                        value: 'value1',
                        start: {
                            offset: 81,
                            line: 7,
                            character: 4,
                        },
                        end: {
                            offset: 89,
                            line: 7,
                            character: 12,
                        },
                    },
                    {
                        kind: 'StringLiteral',
                        literal: "'value2'",
                        value: 'value2',
                        start: {
                            offset: 91,
                            line: 7,
                            character: 14,
                        },
                        end: {
                            offset: 99,
                            line: 7,
                            character: 22,
                        },
                    },
                    null,
                ],
                start: {
                    offset: 75,
                    line: 6,
                    character: 0,
                },
                end: {
                    offset: 102,
                    line: 8,
                    character: 1,
                },
            },
            comment: null,
            start: {
                offset: 18,
                line: 1,
                character: 0,
            },
            end: {
                offset: 102,
                line: 8,
                character: 1,
            },
        },
        rvalue: {
            kind: 'ListLiteral',
            params: [null],
            start: {
                offset: 104,
                line: 9,
                character: 0,
            },
            end: {
                offset: 106,
                line: 9,
                character: 2,
            },
        },
        comment: null,
        start: {
            offset: 18,
            line: 1,
            character: 0,
        },
        end: {
            offset: 106,
            line: 9,
            character: 2,
        },
    },
    rvalue: null,
    comment: null,
    start: {
        offset: 18,
        line: 1,
        character: 0,
    },
    end: {
        offset: 107,
        line: 9,
        character: 3,
    },
};
