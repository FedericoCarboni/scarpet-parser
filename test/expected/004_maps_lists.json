{
  "errors": [],
  "warnings": [
    {
      "diagnostic": 10,
      "start": {
        "pos": 72,
        "row": 5,
        "col": 1
      },
      "end": {
        "pos": 73,
        "row": 5,
        "col": 2
      }
    },
    {
      "diagnostic": 10,
      "start": {
        "pos": 105,
        "row": 9,
        "col": 1
      },
      "end": {
        "pos": 106,
        "row": 9,
        "col": 2
      }
    }
  ],
  "root": {
    "kind": "BinaryExpression",
    "operator": ";",
    "lvalue": {
      "kind": "BinaryExpression",
      "operator": ";",
      "lvalue": {
        "kind": "BinaryExpression",
        "operator": ";",
        "lvalue": {
          "kind": "BinaryExpression",
          "operator": ";",
          "lvalue": {
            "kind": "MapLiteral",
            "params": [
              {
                "kind": "BinaryExpression",
                "operator": "->",
                "lvalue": {
                  "kind": "StringLiteral",
                  "literal": "'key1'",
                  "value": "key1",
                  "start": {
                    "pos": 24,
                    "row": 2,
                    "col": 4
                  },
                  "end": {
                    "pos": 30,
                    "row": 2,
                    "col": 10
                  }
                },
                "rvalue": {
                  "kind": "StringLiteral",
                  "literal": "'value1'",
                  "value": "value1",
                  "start": {
                    "pos": 34,
                    "row": 2,
                    "col": 14
                  },
                  "end": {
                    "pos": 42,
                    "row": 2,
                    "col": 22
                  }
                },
                "comment": null,
                "start": {
                  "pos": 24,
                  "row": 2,
                  "col": 4
                },
                "end": {
                  "pos": 42,
                  "row": 2,
                  "col": 22
                }
              },
              {
                "kind": "BinaryExpression",
                "operator": "->",
                "lvalue": {
                  "kind": "StringLiteral",
                  "literal": "'key2'",
                  "value": "key2",
                  "start": {
                    "pos": 48,
                    "row": 3,
                    "col": 4
                  },
                  "end": {
                    "pos": 54,
                    "row": 3,
                    "col": 10
                  }
                },
                "rvalue": {
                  "kind": "StringLiteral",
                  "literal": "'value2'",
                  "value": "value2",
                  "start": {
                    "pos": 58,
                    "row": 3,
                    "col": 14
                  },
                  "end": {
                    "pos": 66,
                    "row": 3,
                    "col": 22
                  }
                },
                "comment": null,
                "start": {
                  "pos": 48,
                  "row": 3,
                  "col": 4
                },
                "end": {
                  "pos": 66,
                  "row": 3,
                  "col": 22
                }
              },
              null
            ],
            "start": {
              "pos": 18,
              "row": 1,
              "col": 0
            },
            "end": {
              "pos": 69,
              "row": 4,
              "col": 1
            }
          },
          "rvalue": {
            "kind": "MapLiteral",
            "params": [
              null
            ],
            "start": {
              "pos": 71,
              "row": 5,
              "col": 0
            },
            "end": {
              "pos": 73,
              "row": 5,
              "col": 2
            }
          },
          "comment": null,
          "start": {
            "pos": 18,
            "row": 1,
            "col": 0
          },
          "end": {
            "pos": 73,
            "row": 5,
            "col": 2
          }
        },
        "rvalue": {
          "kind": "ListLiteral",
          "params": [
            {
              "kind": "StringLiteral",
              "literal": "'value1'",
              "value": "value1",
              "start": {
                "pos": 81,
                "row": 7,
                "col": 4
              },
              "end": {
                "pos": 89,
                "row": 7,
                "col": 12
              }
            },
            {
              "kind": "StringLiteral",
              "literal": "'value2'",
              "value": "value2",
              "start": {
                "pos": 91,
                "row": 7,
                "col": 14
              },
              "end": {
                "pos": 99,
                "row": 7,
                "col": 22
              }
            },
            null
          ],
          "start": {
            "pos": 75,
            "row": 6,
            "col": 0
          },
          "end": {
            "pos": 102,
            "row": 8,
            "col": 1
          }
        },
        "comment": null,
        "start": {
          "pos": 18,
          "row": 1,
          "col": 0
        },
        "end": {
          "pos": 102,
          "row": 8,
          "col": 1
        }
      },
      "rvalue": {
        "kind": "ListLiteral",
        "params": [
          null
        ],
        "start": {
          "pos": 104,
          "row": 9,
          "col": 0
        },
        "end": {
          "pos": 106,
          "row": 9,
          "col": 2
        }
      },
      "comment": null,
      "start": {
        "pos": 18,
        "row": 1,
        "col": 0
      },
      "end": {
        "pos": 106,
        "row": 9,
        "col": 2
      }
    },
    "rvalue": null,
    "comment": null,
    "start": {
      "pos": 18,
      "row": 1,
      "col": 0
    },
    "end": {
      "pos": 107,
      "row": 9,
      "col": 3
    }
  }
}